src/store/SingletonViews.js

```diff
@@ -0,0 +1,23 @@
+import { observable } from 'mobx';
+import { createObjectFromLocalStorage } from '../utils';
+
+let singletonViews = null;
+
+class SingletonViews {
+    constructor(mainStore) {
+        if (!singletonViews) {
+            singletonViews = this;
+        }
+        this.mainStore = mainStore;
+        this.views = createObjectFromLocalStorage('cq-views') || [];
+        return singletonViews;
+    }
+
+    @observable views = [];
+
+    updateLocalStorage = () => {
+        if (!this.mainStore.state.chartId) return;
```

Comment:
we should still let users save templates even though chart ID is not defined. Otherwise saving template in statements page won't do anything.

Also don't pass mainStore to a singleton. mainStore instance is unique to each smartchart instance, whereas singletons have only one copy for the entire life time of the app.